static/js/components/CommentForms.js

```diff
@@ -65,59 +66,77 @@ const commentForm = (
   autoFocus: boolean,
-  wysiwyg: boolean = false
-) => (
-  <div className="reply-form">
-    <form
-      onSubmit={onSubmit}
-      className="form"
-      onKeyDown={e => {
-        if (e.key === "Enter" && e.ctrlKey && !disabled && !isEmptyText(text)) {
-          onSubmit(e)
+  wysiwyg: boolean = false,
+  onTogglePopup?: Function
+) => {
+  return (
+    <div className="reply-form">
+      <form
+        onSubmit={
+          userIsAnonymous() && onTogglePopup
+            ? R.compose(
+              onTogglePopup,
+              e => e.preventDefault()
+            )
+            : onSubmit
         }
-      }}
-    >
-      <div className="form-item">
-        {wysiwyg ? (
-          <Editor
-            initialValue={text || ""}
-            onChange={editorUpdateFormShim("text", onUpdate)}
-            autoFocus={autoFocus}
-          />
-        ) : (
-          <textarea
-            name="text"
-            type="text"
-            className="input"
-            placeholder="Write a reply here..."
-            value={text || ""}
-            onChange={onUpdate}
-            autoFocus={autoFocus}
-          />
-        )}
-      </div>
-      <button
-        type="submit"
-        className={`blue-button ${disabled ? "disabled" : ""}`}
-        disabled={disabled || isEmptyText(text)}
+        className="form"
+        onKeyDown={e => {
+          if (
+            e.key === "Enter" &&
+            e.ctrlKey &&
+            !disabled &&
+            !isEmptyText(text)
+          ) {
+            onSubmit(e)
+          }
+        }}
       >
-        Submit
-      </button>
-      {isComment ? (
-        <a
-          href="#"
-          onClick={R.compose(
-            cancelReply,
-            e => e.preventDefault()
+        <div className="form-item">
+          {wysiwyg ? (
+            <Editor
+              initialValue={text || ""}
+              onChange={editorUpdateFormShim("text", onUpdate)}
+              autoFocus={autoFocus}
+            />
+          ) : (
+            <textarea
+              name="text"
+              type="text"
+              className="input"
+              placeholder="Write a reply here..."
+              value={text || ""}
+              onChange={disabled ? null : onUpdate}
+              onClick={
+                userIsAnonymous() && onTogglePopup ? onTogglePopup : null
+              }
+              autoFocus={autoFocus}
+            />
           )}
-          className="cancel-button"
+        </div>
+        <button
+          type="submit"
+          className={`blue-button ${disabled ? "disabled" : ""}`}
+          disabled={(disabled || isEmptyText(text)) && !userIsAnonymous()}
         >
-          Cancel
-        </a>
-      ) : null}
-    </form>
-  </div>
-)
+          Submit
+        </button>
+        {isComment ? (
+          <a
+            href="#"
+            onClick={R.compose(
+              cancelReply,
+              e => e.preventDefault()
+            )}
```

Comment:
I think we could update this to use `preventDefaultAndInvoke` as well, while we're in here